features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/Base64LRUCache.kt

```diff
@@ -0,0 +1,99 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.recorder.base64
+
+import android.content.ComponentCallbacks2
+import android.content.res.Configuration
+import android.graphics.drawable.Drawable
+import android.util.LruCache
+import androidx.annotation.VisibleForTesting
+
+@Suppress("MagicNumber")
+internal object Base64LRUCache : Cache, ComponentCallbacks2 {
+    private val CACHE_MEMORY_SIZE = (Runtime.getRuntime().maxMemory() / 8).toInt()
```

Comment:
What is everyone's opinion about these memory thresholds + the eviction strategy for the cache? Should we be less aggressive in evicting - e.g. only evict 50% of entries in critical memory situations? I see that Glide, for example, only evicts 50% when critical.

Reply:
regarding max memory size: That is quite a lot for our cache imo, given that our SDK is not the only citizen in the application process and there are obviously image libraries which create cache as well. Should we maybe go lower?

Also I guess this constant should be called `MAX_CACHE_MEMORY_SIZE_BYTES` (to highlight that it is the max value and also the unit)?.

Regarding eviction strategy: it is not that important, so I guess 50% is okay.

Reply:
With image downscaling it will be possible to use a small cache size and still handle a relatively large number of images, so I've set the cache to 4mb (as we did on ios)